Cesium3DTilesSelection/src/Tileset.cpp

```diff
@@ -53,25 +50,33 @@ Tileset::Tileset(
                     options.credit.value(),
                     options.showCreditsOnScreen))
               : std::nullopt),
-      _url(url),
-      _isRefreshingIonToken(false),
       _options(options),
       _pRootTile(),
       _previousFrameNumber(0),
-      _loadsInProgress(0),
-      _subtreeLoadsInProgress(0),
       _overlays(*this),
-      _tileDataBytes(0),
-      _supportsRasterOverlays(false),
       _gltfUpAxis(CesiumGeometry::Axis::Y),
       _distancesStack(),
       _nextDistancesVector(0) {
   if (!url.empty()) {
     CESIUM_TRACE_USE_TRACK_SET(this->_loadingSlots);
-    this->notifyTileStartLoading(nullptr);
-    LoadTilesetDotJson::start(*this, url).thenInMainThread([this]() {
-      this->notifyTileDoneLoading(nullptr);
-    });
+    TilesetJsonLoader::createLoader(externals, url, {})
+        .thenInMainThread([this](TilesetContentLoaderResult&& result) {
+          if (result.errors) {
+            this->getOptions().loadErrorCallback(TilesetLoadFailureDetails{
```

Comment:
I go with option 1 and 2 in unit test PR. If loadErrorCallback is not specified, it will log error and warning. Otherwise, no log happens in this case and the result will be passed to the callback. I leave this comment here to mark this as temporarily resolved until the PR is merged